09_regexp.md

```diff
@@ -986,71 +696,49 @@ while (match = number.exec(input)) {
 
 {{id ini}}
-## Parsing an INI file
+## 解析一个 INI 文件
 
 {{index comment, "file format", "enemies example", "INI file"}}
 
-To conclude the chapter, we'll look at a problem that calls for
-((regular expression))s. Imagine we are writing a program to
-automatically collect information about our enemies from the
-((Internet)). (We will not actually write that program here, just the
-part that reads the ((configuration)) file. Sorry.) The configuration
-file looks like this:
+现在，我们来试图解决一个需要调用((正则表达式))的问题来总结这一章的内容。试想一下，我们现在打算写一个可以程序，它可以自动收集我们在((互联网))上仇敌的信息。(开玩笑，我们并不会真的在这里写这样一个程序，只是写其中可以读取((配置))文件的部分而已。)以下是我们的配置文件：
 
 ```{lang: "text/plain"}
 searchengine=https://duckduckgo.com/?q=$1
 spitefulness=9.7
 
-; comments are preceded by a semicolon...
-; each section concerns an individual enemy
+; 分号后面的是注释
+; 每一个区块针对一个仇敌
 [larry]
-fullname=Larry Doe
-type=kindergarten bully
+fullname=Larry Doe 拉里·兜
+type=kindergarten bully 幼儿园小霸王
 website=http://www.geocities.com/CapeCanaveral/11451
 
 [davaeorn]
-fullname=Davaeorn
-type=evil wizard
+fullname=Davaeorn 达瓦尔
+type=evil wizard 邪恶巫师
 outputdir=/home/marijn/enemies/davaeorn
 ```
 
 {{index grammar}}
 
-The exact rules for this format (which is a widely used format,
-usually called an _INI_ file) are as follows:
+这个格式(一种很常见的格式，通常被称为 _INI_ 文件)的书写规则如下：
 
-- Blank lines and lines starting with semicolons are ignored.
+- 空白行或者由分号起头的行会被忽略
 
-- Lines wrapped in `[` and `]` start a new ((section)).
+- 被方括号 `[` 和 `]` 括起来的行意味着开始一个新的((区块))
 
-- Lines containing an alphanumeric identifier followed by an `=`
-  character add a setting to the current section.
+- 如果一行含有字母或数字的字符，而且其后缀为 `=` 符号的话，则会为当前的区块增加一条设定。
 
-- Anything else is invalid.
+- 其余的内容都会被视为无效。
 
-Our task is to convert a string like this into an object whose
-properties hold strings for settings written before the first
-section header and subobjects for sections, with those subobjects
-holding the section's settings.
+我们的任务是把类似于上述示例中的字符串给转换成一个对象，这个对象的属性会包含第一个区块标题之前用作设定的字符串，然后它还有针对区块的子对象，这些子对象包含对每一个区块的设定。
 
 {{index "carriage return", "line break", "newline character"}}
 
-Since the format has to be processed ((line)) by line, splitting up
-the file into separate lines is a good start. We saw
-the `split` method in [Chapter ?](data#split).
-Some operating systems, however, use not just a newline character to
-separate lines but a carriage return character followed by a newline
-(`"\r\n"`). Given that the `split` method also allows a regular
-expression as its argument, we can use a regular expression like
-`/\r?\n/` to split in a way that allows both `"\n"` and `"\r\n"`
-between lines.
+由于这种格式的文件会被逐行处理，将其拆分成单独的行是一个不错的开始。我们在 [第四章](data#split) 里已经学到过 `split` 方法。然而，有的操作系统不单用一个换行符来分行，而是用回车符后跟一个换行符 (`"\r\n"`)。鉴于 `split` 方法也可以将正则表达式作为其参数，我们可以用类似于 `/\r?\n/` 的正则表达式来进行分行，它同时允许 `"\n"` 和 `"\r\n"` 这两种方式。
 
 ```{startCode: true}
 function parseINI(string) {
```

Comment:
`// Start with an object to hold the top-level fields` is worth translating
and possibly the strings in `throw new Error("Line '" + line + "' is not valid.");`